apps/mull-api/src/app/post/post.resolver.ts

```diff
@@ -0,0 +1,37 @@
+import { Args, Int, Mutation, Query, Resolver, Subscription } from '@nestjs/graphql';
+import { Post } from '../entities';
+import { CreatePostInput, UpdatePostInput } from './inputs/post.input';
+import pubSub from './post.pubsub';
+import { PostService } from './post.service';
+
+@Resolver(() => Post)
+export class PostResolver {
+  constructor(private readonly postService: PostService) {}
+  @Subscription(/* istanbul ignore next */ () => Post)
+  postAdded() {
+    return pubSub.asyncIterator(['postAdded']);
+  }
+
+  @Mutation(/* istanbul ignore next */ () => Post)
+  async post(@Args('post') post: CreatePostInput) {
+    pubSub.publish('postAdded', {
+      postAdded: post,
+    });
+    return this.postService.createPost(post);
```

Comment:
⚠⚠⚠ As discussed on slack, I think sending a notification to the subscriber before the post was saved successfully can be problematic:
- We don't get an id, since post here is a `CreatePostInput`. Ritchelle was confused as to why she couldn't get the id of the new post
- What if the post doesn't save successfully? Then the sub got a wrong notification, since the db didn't change.

Reply:
Good catch, I'll push a quick fix